File: jimmer-ddd-java-sample/domain/src/main/java/org/morecup/jimmerddd/java/sample/domain/order/OrderFactory.java
package org.morecup.jimmerddd.java.sample.domain.order;

import org.morecup.jimmerddd.java.sample.domain.order.dto.CreateOrderCmd;
import org.springframework.stereotype.Service;

@Service
public class OrderFactory{
    private final OrderRepository orderRepository;

    public OrderFactory(OrderRepository orderRepository) {
        this.orderRepository = orderRepository;
    }

    public Order create(CreateOrderCmd cmd) {
        return cmd.toEntity();
    }
}

File: jimmer-ddd-java-sample/domain/src/main/java/org/morecup/jimmerddd/java/sample/domain/order/OrderFactory.kt
package org.morecup.jimmerddd.kotlin.sample.domain.order

import org.morecup.jimmerddd.core.factory.autoContext
import org.morecup.jimmerddd.kotlin.sample.domain.order.dto.CreateOrderCmd
import org.springframework.stereotype.Service

@Service
class OrderFactory(
    private val orderRepository: OrderRepository
) {
    fun create(cmd: CreateOrderCmd): Order = autoContext{
        cmd.toEntity()
    }
}
